src/triage/component/catwalk/evaluation.py

```diff
@@ -209,12 +264,20 @@ def _evaluations_for_threshold(
                 raise metrics.UnknownMetricError()
 
             for parameter_combination in parameters:
-                value = self.available_metrics[metric](
-                    predictions_proba,
-                    predicted_classes_with_labels,
-                    present_labels,
-                    parameter_combination,
-                )
+                try:
+                    value = self.available_metrics[metric](
+                        predictions_proba,
+                        predicted_classes_with_labels,
+                        present_labels,
+                        parameter_combination,
+                    )
+                except (ValueError, IndexError) as e:
+                    warnings.warn(
+                        f"{metric} not defined for parameter "
+                        "{parameter_combination}. Inserting NULL for value.",
+                        RuntimeWarning
+                    )
```

Comment:
there is _some_ place for warnings, but it's a pretty annoying module to work with, and I mostly see it limited to deprecation notices.

is this definitely a case wherein we want to continue the run? or does it make sense to fail and thereby ensure the runner sees the message?

if we want to run, would it be appropriate to use `logging`?

for example, say you were running this on aws-batch -- can you be sure that anyone will ever get this information?

Reply:
This could happen in two cases:

- all of the labels above some threshold are the same (ValueError), so some metrics are undefined because they result in division by 0. We want to continue because this is expected to happen occasionally, especially at small thresholds
- a subset is empty for a specific matrix (IndexError); the pipeline will already error out if the subset table is empty for every as_of_date, so I don't think we want to error here but continue

I imagine someone seeing this if these if they notice that some metrics they were using are missing and then check the logs to find out why. Perhaps it makes sense to catch the errors separately and put more useful messages (with logging.warn).

Reply:
Sounds good to me.